lib/services/ngsi/entities-NGSI-LD.js

```diff
@@ -383,6 +383,56 @@ function sendQueryValueNgsiLD(entityName, attributes, typeInformation, token, ca
     );
 }
 
+/**
+ * Adds any linked data items to the NGSI-LD payload for the context broker.
+ * For example a Building entity linked to a temperature gauge Device will
+ * also be updated as measures are received to update the Device Entity.
+ *
+ * @param {Object} typeInformation  Configuration information for the device.
+ * @param {Object} json             Payload to send to the context broker
+ */
+function addLinkedEntities(typeInformation, json) {
+    // use the supplied type if possible, but fallback to reading the
+    // type from the id of the entity.
+    function getEntityType(id, target) {
+        if (target.type) {
+            return target.type;
+        }
+        let calculatedType = id.replace(NGSI_LD_URN, '');
```

Comment:
as it is based on a recommendation from the specification, maybe it worth adding a log message (something like `warning, fallback to guessing the type from id as no type attribute was found ...`)?

Reply:
This substitution is already made silently in two places:

* https://github.com/telefonicaid/iotagent-node-lib/blob/2835d6d50a27a2406ca316568c8ea60b4f7cddde/lib/services/devices/registrationUtils.js#L467

* https://github.com/telefonicaid/iotagent-node-lib/blob/a7eb2605d342ed2b50821d92f967d93407e90e65/lib/services/ngsi/entities-NGSI-LD.js#L198

The question to @fgalan and the other owners of the repo is whether this should the codebase:

*  have logging in all three places, for consistency 
*  have no logging for any of these instances 

Once this is clear I'll follow up either way.

Reply:
I gusss that a debug trace could be useful and doesn't hurt... so the "have debug logging in all three places, for consistency" option should be the better one.

Reply:
Fixed 64c7eae

Reply:
LGTM